craft_application/commands/base.py

```diff
@@ -48,9 +45,20 @@ def __init__(self, config: dict[str, Any] | None) -> None:
         self._app: application.AppMetadata = config["app"]
         self._services: service_factory.ServiceFactory = config["services"]
 
+    def run_managed(
+        self,
+        parsed_args: argparse.Namespace,  # noqa: ARG002 (the unused argument is for subclasses)
+    ) -> bool:
+        """Whether this command should run in managed mode.
+
+        By default returns `False`. Subclasses can override this method to change this,
+        including by inspecting the arguments in `parsed_args`.
+        """
+        return False
```

Comment:
I like this for hybrid commands, but it seems like a lot of text for the more common case of managed commands.

I've thought of two options for this, both of which are variations on this method.

1. Similar to what you have in the tests below, make a class-level `_run_managed` that we can use for managed/unmanaged commands. (Perhaps default it to `None` and have the `AppCommand` implementation raise an exception if it's `None` so we have to explicitly make it managed, unmanaged, or override `run_managed` for a hybrid command?)
2. Make this an abstract method and then subclass `AppCommand` as `ManagedCommand` and `UnmanagedCommand` that provide `final` implementations of `run_managed`. (Hybrid command trees would then inherit directly from `AppCommand` and have to implement `run_managed` themselves, but the common use case would inherit either a managed command or an unmanaged command).

Reply:
Well I can remove the test if you feel its too noisy, but in my view once we support destructive mode the majority of the commands (the lifecycle ones) will actually be hybrid. Then the remaining commands (init, version, etc) that an app typically has are going to be non-managed and can use this default, no?